### PR TITLE
Fix/bulk edit journals

### PR DESCRIPTION
**Pull request by marius-balteanu**

### CODE REVIEW

**Comment by marius-balteanu** [comment]
@mayenok, @PowerKiKi, I've added the migration script that should remove all wrong journals generated by this bug. Do you have the possibility to test the migration script in your environments just to double check before merging?

**Comment by PowerKiKi** [comment]
I don't have non-production env at the moment. Does someone else is able to test it ?

**Comment by mayenok** [comment]
@marius-balteanu, I have checked on my environment: the first migration works just fine. As for the second - in the JournalDetail table I have over 1,7 million rows but I have no entry that satisfies the condition:

> .where("#{JournalDetail.table_name}.journal_id IS NULL")

Maybe you misspelled it or you've meant something else?

**Comment by marius-balteanu** [comment]
That condition should return only the rows from journals table that have no relation in the journal_details table. The scope of the second migration is to find the entries from the journal table that are not notes (the first condition) and in the same time, don’t have any attribute change (entry in journal_details). 

A scenario for this case is the following:
1. bulk edit at least two issues that have a common tag but without changing anything (just open the bulk edit page and press the save button)
2. an entry is saved in the journals table 
3. an entry is saved in the journal_details table which records the wrong tag list changes

The first migration script will remove the entry from the journal_detail table (3). In that moment, the entry from the journals table (2) remains incorrect because it doesn't record any notes or attribute change. In the UI, the journal will not be shown, but I still think that it is better to remove the inconsistencies from the db. 

In conclusion, it is ok to return nothing if you don't have any entry in the above case or you didn't run the first migration and you just run the command in the console without calling the destroy_all method.
